include/oneapi/dpl/pstl/utils_ranges.h

```diff
@@ -87,21 +189,25 @@ using __range_size_t = typename __range_size<_R>::type;
 
 template <typename _R>
 auto
-__check_size(int) -> decltype(std::declval<_R&>().size());
+__check_size(int) -> decltype(__get_buffer_size(std::declval<_R&>()));
```

Comment:
Is it `__dpl_sycl::__get_buffer_size`? This is the only one I find in the library, but this code should not rely on SYCL utilities - it's not guarded by the corresponding macro.